red-hat-data-services/odh-manifests
# Update delegate urls argument for dashboard oauth

There was a misconfiguration in the --openshift-delegate-urls parameter
for the `authentication` overlay of the dashboard. This caused
authentication/authorization to fail for some use cases, particularly
when attempting to authenticate to the dashboard programmatically using
a bearer token. This change updates the argument to a valid value to fix
this use case.

## Testing instructions

* Deploy RHODS version 1.14
* Run the following command to attempt to log into the dashboard using the `rhods-dashboard` service account. The command should output `403`, indicating that the service account has been restricted from accessing the dashboard:
```
curl --header "Authorization: Bearer $(oc get secret -n redhat-ods-applications -o jsonpath='{.data.token}' `oc get secret -n redhat-ods-applications | grep rhods-dashboard-token | awk '{print $1}'` | base64 -d)" -s -o /dev/null -w "%{http_code}\n" -k https://$(oc get route -n redhat-ods-applications rhods-dashboard -o jsonpath='{.spec.host}')
```
* Deploy this fix using the olminstall setup.sh script and the following catalog source image: quay.io/accorvin/rhods-operator-live-catalog:1.0.0-rhods-4921
* Wait until the RHODS dashboard pods redeploy
* Run the curl command again. This time it should print a status code of `200`, indicating that the service account has successfully reached the dashboard:
```
curl --header "Authorization: Bearer $(oc get secret -n redhat-ods-applications -o jsonpath='{.data.token}' `oc get secret -n redhat-ods-applications | grep rhods-dashboard-token | awk '{print $1}'` | base64 -d)" -s -o /dev/null -w "%{http_code}\n" -k https://$(oc get route -n redhat-ods-applications rhods-dashboard -o jsonpath='{.spec.host}')
```
* Navigate to the rhods dashboard in a browser. Confirm that you can log in and use the dashboard as normal.

- [x] JIRA link(s): https://issues.redhat.com/browse/RHODS-4921
